src/qiskit_qasm3_import/expression.py

```diff
@@ -286,15 +315,15 @@ def visit_IndexedIdentifier(self, node: ast.IndexedIdentifier):
 
 
 def resolve_condition(
-    node: ast.Expression, symbols: Mapping[str, Symbol]
+    node: ast.Expression, context: "State"
```

Comment:
Not sure why the quotes are necessary here - `State` is defined.

Reply:
Last time I tried, I got a "partially initialized module" (or something like that) error.

Reply:
Specifically,  removing the quotes and  importing `State` as in this [modification of expression.py](https://github.com/jlapeyre/qiskit-qasm3-import/blob/remove-quotes-from-state/src/qiskit_qasm3_import/expression.py) gives
```
ImportError: cannot import name 'State' from partially initialized module 'qiskit_qasm3_import.converter' (most likely due to a circular import) (/home/lapeyre/qiskit/qiskit-qasm3-import/src/qiskit_qasm3_import/converter.py)```

There may be a way around this. I did not yet try.

Reply:
Moving `State` to a new file `state.py` is a possibility. I made this change in [another branch](https://github.com/jlapeyre/qiskit-qasm3-import/tree/remove-quotes-from-state/src/qiskit_qasm3_import) and all tests pass.

Reply:
In an earlier commit, I introduced `state.py` to this PR. This takes care of the problem here. Resolving converstation.